### PR TITLE
test(ctb): add test for stuck ETH on failed msg

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds a test that ETH gets properly stuck in the CrossDomainMessenger
contracts but then is unstuck when the message is replayed.

### CODE REVIEW

File: packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol [hunk 1]
```diff
@@ -216,4 +222,54 @@ contract L1CrossDomainMessenger_Test is Messenger_Initializer {
...
+            target,
+            value,
+            0,
+            hex"1111"
+        );
+
+        assertEq(address(L1Messenger).balance, value);
+        assertEq(address(target).balance, 0);
+        assertEq(L1Messenger.receivedMessages(hash), true);
+
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol:254]: Could also add assertion on successful messages being false

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
```

Learn more at https://hardhat.org/reference/solidity-support

Error HH701: There are multiple artifacts for contract "Reverter", please use a fully qualified name.

Please replace Reverter for one of these options wherever you are trying to read its artifact:

L1CrossDomainMessenger.t.sol:Reverter
L2CrossDomainMessenger.t.sol:Reverter
```

File: packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol [hunk 2]
```diff
@@ -215,4 +215,55 @@ contract L1CrossDomainMessenger_Test is Messenger_Initializer {
...
+            target,
+            value,
+            0,
+            hex"1111"
+        );
+
+        assertEq(address(L1Messenger).balance, value);
+        assertEq(address(target).balance, 0);
+        assertEq(L1Messenger.successfulMessages(hash), false);
+        assertEq(L1Messenger.receivedMessages(hash), true);
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol:247]: Can we rename this to `replayableMessages`?

**Review by maurelian — APPROVED** [review]

File: packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol [hunk 3]
```diff
@@ -215,4 +215,55 @@ contract L1CrossDomainMessenger_Test is Messenger_Initializer {
...
+        address sender = PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER;
+        uint256 value = 100;
+
+        vm.expectCall(target, hex"1111");
+
+        bytes32 hash = Hashing.hashCrossDomainMessage(0, sender, target, value, 0, hex"1111");
+
+        uint256 senderSlotIndex = 51;
+        vm.store(address(op), bytes32(senderSlotIndex), bytes32(abi.encode(sender)));
+        vm.etch(target, address(new Reverter()).code);
```
[the remark this answers is not in the document]
Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol:232]: Not important enough to block merging.

**Pull request merged**